src/balancer/accept_http.rs

```diff
@@ -279,15 +266,22 @@ async fn forward_body(
     head_cache: &Arc<RwLock<BTreeMap<u64, Vec<String>>>>,
     cache: Arc<Db>,
     params: RequestParams,
+    config: &Arc<RwLock<Settings>>
 ) -> (
     Result<hyper::Response<Full<Bytes>>, Infallible>,
     Option<usize>,
 ) {
-    // Check if body has application/json
-    let mut tx = tx;
-    if tx.headers().get("content-type") != Some(&HeaderValue::from_static("application/json")) {
-        let headers = tx.headers_mut();
-        headers.insert("content-type", HeaderValue::from_static("application/json"));
+    if config.read().unwrap().header_check {
```

Comment:
would be a good idea to grab the value of `header_check` once when initially receiving the request so we dont have to keep acquiring locks on it. probably should be a part of the refactor mentioned below?